src/server/debugcmd.cc

```diff
@@ -194,6 +195,28 @@ void DoPopulateBatch(string_view type, string_view prefix, size_t val_size, bool
       stub_tx->InitByArgs(cntx->ns, local_cntx.conn_state.db_index, args_span);
 
       sf->service().InvokeCmd(cid, args_span, &crb, &local_cntx);
+
+      if (expire_ttl_range.has_value()) {
```

Comment:
Hi @lichuang 
Thank you for contributing!
I see there is a bug here.
This if should be outside the while(elements_left) loop 
Inside the while loop we continue to populate the same key, so when you do this in the while loop you run expire multiple times for the same key and I see that you also use std::move(key) below which actually breaks things. Dont know why it did not failed in our tests..
Would you like to follow up on this and provide a fix?

Reply:
ok, i take care of this issue now

Reply:
https://github.com/dragonflydb/dragonfly/pull/4757